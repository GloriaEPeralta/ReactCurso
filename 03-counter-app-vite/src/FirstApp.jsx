import Proptypes from 'prop-types';


function Encabezado ()  {
    return 'Encabezado de la aplicación';
}   

export const FirstApp = ({title,subtitle, name}) => {
    
    return (
        <>
        <h1>{Encabezado()} </h1>      
        {<h2>{title}</h2>}
        <p>{subtitle}</p>
        <p>{name}</p>   
        </>
    );
    }   

FirstApp.propTypes = {
    title: Proptypes.string.isRequired,
    subtitle: Proptypes.string.isRequired  
}
FirstApp.defaultProps = {  
   
    subtitle: 'Soy un subtítulo',
    name: 'Katy Peralta'
}

