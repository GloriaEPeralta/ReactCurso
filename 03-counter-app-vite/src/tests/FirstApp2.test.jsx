import { render,screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";


describe('Pruebas en <FirstApp />', () => 
{ 
  const title='Soy un titulo';    
  test('La primera prueba deber hacer match con el snapshot', () => 
  {  
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();    
  });

  test('Debe mostrar el mensaje "Soy un Titulo"', () => 
    {  
       render(<FirstApp title={title} />);
      // expect(screen.getByText(title)).toBeTruthy();  
      //screen.debug();
      expect(screen.getByRole('heading',{level:2}).innerHTML).toContain(title); 
    });

    // hizo sobre subtitulo pero ya ES MUCHO!! NO LO HICE
});
