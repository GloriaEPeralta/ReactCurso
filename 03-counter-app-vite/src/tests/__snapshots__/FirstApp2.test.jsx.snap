// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> La primera prueba deber hacer match con el snapshot 1`] = `
<div>
  <h1>
    Encabezado de la aplicación
     
  </h1>
  <h2>
    Soy un titulo
  </h2>
  <p>
    Soy un subtítulo
  </p>
  <p>
    Katy Peralta
  </p>
</div>
`;
